proposals/0000-improved-ghc-development.rst

```diff
@@ -0,0 +1,216 @@
+
+  We could have harbormaster build the docs and provide them to trac. I am not sure how reviews would go. Styling would be another issue. 
+
+  Lastly this plugin seems to only support SVN. but that's not a huge barrier.
+
+- Helping Newcomers
+
+   * Create a better landing page for newcomers (see proposal below)
+   * Show newcomers tickets more prominently
+   * Clean up build instructions. The pages have become way too complex over the years. 
+   * Create native setup script particular to an environment.
+
+  As I am a windows maintainer, I will offer a proposal: Use scripts similar to my own setup scripts. my setup script for a 100% unattended build env setup for Windows are here: https://github.com/Mistuke/GhcDevelChoco/releases
+
+  These are entirely self contained environments that can be removed by a simple rm -rf /.
+  You can have as many as you want on the same machine without them interfering with eachother or with whatever else you might have done to your GHC already installed.
+   
+  It's not 100% production ready but it works and does so well.
+
+  This script will setup an environment containing (if you want):
+   * A full msys setup
+   * gdb/gcc etc for debugging
+   * check out Haskell sources
+   * do a local install of php and arc
+   * configure arc
+   * setup Hadrian
+   * setup SSH (for those who want to interface with it this way)
+ 
+  The script is set up to provide you with a working environment with little interaction needed. Setup is just one call. 
+
+  For Windows, such a script is a much better solution than Docker, simply because Docker does not work very well for Windows targets.
+
+- Phab reviewers list
+
+  Reviewers are currently assigned based on a set of Phabricator rules. As a maintainer on a less popular platform I know the pain of getting people to review your code.
+  
+  So my suggestion is to also assign reviewers based on the git history of the files you're changing. 
+
+  The reason for this is that currently it's always the same people reviewing patches.
+  Their time is spread thin. Particularly on less popular platforms it basically comes down to 4 people.
+  
+- Update trac linters
+
+  The git linters and Arc linters aren't currently checking the same things.
+
+  Particularly is the fact that the pre-commit hooks check
+  the summary text but arc doesn't. So commits can get rejected at push time requiring more fork for maintainers.
+
+  Also I want to say I love the summary document you have to fill in.
+  It ensures useful information is there later when I have to find out why a change was made. So whatever we do, don't remove this.
+
+- Phab signup screen
+
+  It's recently been made a requirement to require a public key to push to phab. The error you get when you don't do this and try to push a patch is very very cryptic and unintuitive. Could we make a plugin that asks the user to upload a public key on trac if they haven't done so? Like a banner at the top?
```

Comment:
Why would we make it part of Trac? Then how would we ever get it into Phabricator?

Either way, in theory we could add another conduit API endpoint that would actually allow updating a user's SSH keys, maybe. You can retrieve them right now, but not create them. In theory though, we could have a plugin to `arc` that could do something like this, where it could upload a key in `~/.ssh/id_rsa.pub` to your account when you ask it to.